src/hooks/useGlobalStore.tsx

```diff
@@ -0,0 +1,87 @@
+import {createContext, useContext, useMemo, useState} from 'react';
+import {CommentData, EntityPointer, EntityType, FileData, FullProjectData, SectionData} from '../types';
+import {matchesEntityPointer} from '../utils';
+
+type GlobalStoreContextValue = {
+    getFullProjectData(): FullProjectData;
+    setFullProjectData(project: FullProjectData): void;
+}
+
+const globalStoreContext = createContext<GlobalStoreContextValue | null>(null);
+
+type GlobalStoreProviderProps = React.PropsWithChildren<{
+    initialProjectData: FullProjectData;
+}>;
+
+type UseGlobalStoreHookValue = {
+    getCommentsForSection(sectionId: string): CommentData[];
+    getCommentsForFile(fileId: string): CommentData[];
+    getFilesForSection(sectionId: string): FileData[];
+    getCommentsForEntity(entityPointer: EntityPointer): CommentData[];
+    getSection(sectionId: string): SectionData;
+
+    addComment(comment: CommentData): void;
+    updateSection(sectionId: string, section: SectionData): void;
+}
```

Comment:
The functions defined in this file allow the components to get data stored in the React context. The "Global Store" is the place where data that has been already fetched using the client is available. It's essentially a big `useState` for all user-generated data